Dockerfile

```diff
@@ -2,11 +2,16 @@
 FROM python:3.9
 
 # Set working directory.
+WORKDIR /assignments/git-docker/
 
 # Expose port 8000 for your FastAPI server.
-
-# Install Python requirements using Pip.
+EXPOSE 8000
 
 # Copy your codebase into the Docker container.
+COPY . .
+
+# Install Python requirements using Pip.
+RUN pip install -r requirements.txt
```

Comment:
Your Dockerfile is fine overall. However, it's better practice to install requirements before copying the codebase (by first copying only the requirements.txt file and running pip install).

This is because Docker images are created in layers, with each command being treated as a new layer, and images are cached in a manner so that if you rebuild your Docker image, then the layers can be re-used up until the first point where something in a layer has changed ([reference](https://stackoverflow.com/a/34399661)).

Reply:
Hi, I changed my Dockerfile based on your review, first copying just the requirements.txt file, then installing it, and then copying the whole codebase.